akka-http-caching/src/main/scala/akka/http/caching/Cache.scala

```diff
@@ -0,0 +1,76 @@
+package akka.http.caching
+
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+import scala.util.control.NonFatal
+
+/**
+ * General interface implemented by all akka-http cache implementations.
+ */
+trait Cache[V] { cache ⇒
+
+  /**
+   * Selects the (potentially non-existing) cache entry with the given key.
+   */
+  def apply(key: Any) = new Keyed(key)
+
+  class Keyed(key: Any) {
```

Comment:
Do we still remember why this (somewhat uncommon) code structure was used? In which way is it better then simply providing a few more overloads for `Cache.apply`?